src/deployment/deploy.py

```diff
@@ -287,78 +286,99 @@ def setup_rbac(self) -> None:  # noqa: C901
-                description="Allow access from a lab machine.",
-                value=OnefuzzAppRole.ManagedNode.value,
-            ),
+            {
+                "allowedMemberTypes": ["Application"],
+                "description": "Allows access from the CLI.",
+                "displayName": OnefuzzAppRole.CliClient.value,
+                "id": str(uuid.uuid4()),
+                "isEnabled": True,
+                "value": OnefuzzAppRole.CliClient.value,
+            },
+            {
+                "allowedMemberTypes": ["Application"],
+                "description": "Allow access from a lab machine.",
+                "displayName": OnefuzzAppRole.ManagedNode.value,
+                "id": str(uuid.uuid4()),
+                "isEnabled": True,
+                "value": OnefuzzAppRole.ManagedNode.value,
+            },
         ]
 
-        app: Optional[Application] = None
-
-        if not existing:
+        if not app:
             logger.info("creating Application registration")
 
-            params = ApplicationCreateParameters(
-                display_name=self.application_name,
-                identifier_uris=[self.get_identifier_url()],
-                reply_urls=[self.get_instance_url() + "/.auth/login/aad/callback"],
-                optional_claims=OptionalClaims(id_token=[], access_token=[]),
-                required_resource_access=[
-                    RequiredResourceAccess(
-                        resource_access=[
-                            ResourceAccess(id=USER_READ_PERMISSION, type="Scope")
+            params = {
+                "displayName": self.application_name,
+                "identifierUris": [self.get_identifier_url()],
+                "signInAudience": self.get_signin_audience(),
+                "appRoles": app_roles,
+                "api": {
+                    "oauth2PermissionScopes": [
+                        {
+                            "adminConsentDescription": f"Allow the application to access {self.application_name} on behalf of the signed-in user.",
+                            "adminConsentDisplayName": f"Access {self.application_name}",
+                            "id": str(uuid.uuid4()),
+                            "isEnabled": True,
+                            "type": "User",
+                            "userConsentDescription": f"Allow the application to access {self.application_name} on your behalf.",
+                            "userConsentDisplayName": f"Access {self.application_name}",
+                            "value": "user_impersonation",
+                        }
+                    ]
+                },
+                "web": {
+                    "implicitGrantSettings": {
+                        "enableAccessTokenIssuance": False,
+                        "enableIdTokenIssuance": True,
+                    },
+                    "redirectUris": [
+                        f"{self.get_instance_url()}/.auth/login/aad/callback"
```

Comment:
i moved `get_identifier_url()` not because of the string interpolation but because the value depends on the configuration of the deployment. The redirect uri only depends on the instance_url